src/mappings/shared/token/setMetadata.ts

```diff
@@ -0,0 +1,32 @@
+import { getOptional, getWith } from '@kodadot1/metasquid/entity'
+import { Context } from '../../utils/types'
+import { CollectionEntity as CE, NFTEntity as NE, TokenEntity as TE } from '../../../model'
+import { debug, warn } from '../../utils/logger'
+import { OPERATION, generateTokenId, mediaOf } from './utils'
+import { TokenAPI } from './tokenAPI'
+
+export async function setMetadataHandler(context: Context, collection: CE, nft: NE): Promise<TE | undefined> {
```

Comment:
```suggestion
export async function updateTokenMetada(context: Context, collection: CE, nft: NE): Promise<TE | undefined> {
```

Reply:
here too

it doesn't up[date the token metadata

it removed the nft from the token
and creates a new one or adds it to a previously exiting token

basically move the nft (on which the metadata changed) from 1 token to another